packages/@ember/-internals/utils/lib/mandatory-setter.ts

```diff
@@ -105,7 +105,7 @@ if (DEBUG) {
     if (setters !== undefined && setters[keyName] !== undefined) {
       Object.defineProperty(obj, keyName, setters[keyName]);
 
-      setters[keyName] = undefined;
+      delete setters[keyName];
```

Comment:
Note: I *think* we probably don’t mind because this is in debug-only code, and it’s totally reasonable from a satisfying-the-compiler POV, but this is apt to throw the runtime into deopt mode for the `setters` object, because `delete` totally messes up the shape of the object and the VM (AFAIU) has to basically go into "treat it like a dictionary" mode instead of treating it like a class with a known shape. Most likely this is *already* happening in this spot given all the shenanigans involved with this debug feedback, but worth noting.